tests/kv_test.go

```diff
@@ -0,0 +1,344 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+	"github.com/xline-kv/go-xline/client"
+	"github.com/xline-kv/go-xline/xlog"
+	"go.uber.org/zap/zapcore"
+)
+
+func TestKv(t *testing.T) {
+	xlog.SetLevel(zapcore.WarnLevel)
+
+	curpMembers := []string{"172.20.0.3:2379", "172.20.0.4:2379", "172.20.0.5:2379"}
+
+	client, err := client.Connect(curpMembers)
+	assert.NoError(t, err)
+	kvClient := client.Kv
+
+	t.Run("put", func(t *testing.T) {
```

Comment:
We could split these put/range/etc to multiple test functions for better readibility.

Reply:
I think the readability is fine so far.
```bash
$ GOLANG_PROTOBUF_REGISTRATION_CONFLICT="ignore" go test ./tests/ -v
=== RUN   TestKv
=== RUN   TestKv/put
=== RUN   TestKv/put/overwrite_with_prev_key
=== RUN   TestKv/put/overwrite_again_with_prev_key
=== RUN   TestKv/range
=== RUN   TestKv/range/get_Key
=== RUN   TestKv/range/get_from_key
=== RUN   TestKv/range/get_prefix_keys
=== RUN   TestKv/delete
=== RUN   TestKv/delete/delete_key
=== RUN   TestKv/delete/delete_a_range_of_keys
=== RUN   TestKv/txn
=== RUN   TestKv/txn/transaction_1
=== RUN   TestKv/txn/transaction_2
=== RUN   TestKv/compact
--- PASS: TestKv (0.25s)
    --- PASS: TestKv/put (0.02s)
        --- PASS: TestKv/put/overwrite_with_prev_key (0.00s)
        --- PASS: TestKv/put/overwrite_again_with_prev_key (0.02s)
    --- PASS: TestKv/range (0.03s)
        --- PASS: TestKv/range/get_Key (0.01s)
        --- PASS: TestKv/range/get_from_key (0.02s)
        --- PASS: TestKv/range/get_prefix_keys (0.00s)
    --- PASS: TestKv/delete (0.07s)
        --- PASS: TestKv/delete/delete_key (0.03s)
        --- PASS: TestKv/delete/delete_a_range_of_keys (0.04s)
    --- PASS: TestKv/txn (0.05s)
        --- PASS: TestKv/txn/transaction_1 (0.03s)
        --- PASS: TestKv/txn/transaction_2 (0.01s)
    --- PASS: TestKv/compact (0.07s)
PASS
ok      github.com/xline-kv/go-xline/tests      0.251s
```

Reply:
Thank you for your feedback. I've updated the pull request based on your suggestions. The output now looks like:
```bash
$ GOLANG_PROTOBUF_REGISTRATION_CONFLICT="ignore" go test -coverpkg=./client -covermode=atomic ./tests/ -v
=== RUN   TestPut
=== RUN   TestPut/overwrite_with_prev_key
=== RUN   TestPut/overwrite_again_with_prev_key
--- PASS: TestPut (0.02s)
    --- PASS: TestPut/overwrite_with_prev_key (0.01s)
    --- PASS: TestPut/overwrite_again_with_prev_key (0.00s)
=== RUN   TestRange
=== RUN   TestRange/get_Key
=== RUN   TestRange/get_from_key
=== RUN   TestRange/get_prefix_keys
--- PASS: TestRange (0.03s)
    --- PASS: TestRange/get_Key (0.01s)
    --- PASS: TestRange/get_from_key (0.02s)
    --- PASS: TestRange/get_prefix_keys (0.00s)
=== RUN   TestDelete
=== RUN   TestDelete/delete_key
=== RUN   TestDelete/delete_a_range_of_keys
--- PASS: TestDelete (0.05s)
    --- PASS: TestDelete/delete_key (0.03s)
    --- PASS: TestDelete/delete_a_range_of_keys (0.02s)
=== RUN   TestTxn
=== RUN   TestTxn/transaction_1
=== RUN   TestTxn/transaction_2
--- PASS: TestTxn (0.05s)
    --- PASS: TestTxn/transaction_1 (0.03s)
    --- PASS: TestTxn/transaction_2 (0.02s)
=== RUN   TestCompact
--- PASS: TestCompact (0.07s)
PASS
coverage: 71.3% of statements in ./client
ok      github.com/xline-kv/go-xline/tests      0.225s  coverage: 71.3% of statements in ./client
```